lib/viewmodels/map_view_model.dart

```diff
@@ -1,20 +1,67 @@
 import "dart:async";
+import "package:flutter/material.dart";
 import "package:google_maps_flutter/google_maps_flutter.dart";
 import "package:hooks_riverpod/hooks_riverpod.dart";
 import "package:proxima/models/ui/map_info.dart";
+import "package:proxima/services/geolocation_service.dart";
+import "package:proxima/viewmodels/home_view_model.dart";
 import "package:proxima/views/option_widgets/map/map_selection_option.dart";
 
 // TODO: For now, this is code with mock data.
 // This will just be replaced when we implement a real view model anyway.
-class MapViewModel extends AsyncNotifier<MapInfo> {
+class MapViewModel extends AutoDisposeAsyncNotifier<MapInfo> {
   @override
   Future<MapInfo> build() async {
-    return const MapInfo(
-      currentLocation: LatLng(46.519653, 6.632273),
+    final actualLocation =
+        await ref.read(geoLocationServiceProvider).getCurrentPosition();
+    return MapInfo(
+      currentLocation:
+          LatLng(actualLocation.latitude, actualLocation.longitude),
       selectOption: MapSelectionOptions.nearby,
     );
   }
 
+  //set of circles drawn on the map
+  final Set<Circle> _circles = {};
```

Comment:
Should this not be part of `MapInfo` instead ?

Reply:
I added the set of circles to mapInfo. I still need the getter for the test tho. Fixed with c73c730